op-node/benchmarks/batchbuilding_test.go

```diff
@@ -19,52 +19,81 @@ const (
 	targetOuput_real = uint64(780120)
 )
 
-var (
-	// compressors used in the benchmark
-	rc, _ = compressor.NewRatioCompressor(compressor.Config{
-		TargetOutputSize: targetOutput_huge,
-		ApproxComprRatio: 0.4,
-	})
-	sc, _ = compressor.NewShadowCompressor(compressor.Config{
-		TargetOutputSize: targetOutput_huge,
-	})
-	nc, _ = compressor.NewNonCompressor(compressor.Config{
-		TargetOutputSize: targetOutput_huge,
-	})
-	realsc, _ = compressor.NewShadowCompressor(compressor.Config{
-		TargetOutputSize: targetOuput_real,
-	})
+// compressorDetails is a helper struct to create compressors or supply the configuration for span batches
+type compressorDetails struct {
+	name         string
+	compressorFn func(compressor.Config) (derive.Compressor, error)
+	config       compressor.Config
+}
 
-	// compressors used in the benchmark mapped by their name
-	// they come paired with a target output size so span batches can use the target size directly
-	compressors = map[string]compressorAndTarget{
-		"NonCompressor":        {nc, targetOutput_huge},
-		"RatioCompressor":      {rc, targetOutput_huge},
-		"ShadowCompressor":     {sc, targetOutput_huge},
-		"RealShadowCompressor": {realsc, targetOuput_real},
-	}
+func (cd compressorDetails) String() string {
+	return fmt.Sprintf("%s-%s-%d", cd.name, cd.config.CompressionAlgo, cd.config.TargetOutputSize)
+}
+func (cd compressorDetails) Compressor() (derive.Compressor, error) {
+	return cd.compressorFn(cd.config)
+}
+
+var (
 	// batch types used in the benchmark
 	batchTypes = []uint{
 		derive.SpanBatchType,
-		// uncomment to include singular batches in the benchmark
-		// singular batches are not included by default because they are not the target of the benchmark
-		// derive.SingularBatchType,
 	}
-)
 
-type compressorAndTarget struct {
-	compressor   derive.Compressor
-	targetOutput uint64
-}
+	compAlgos = []derive.CompressionAlgo{
+		derive.Zlib,
+		derive.Brotli,
```

Comment:
[`Brotli` defaults to `Brotli10`](https://github.com/ethereum-optimism/optimism/blob/e81a35a4b864256f15df6c9a064c1dfadb30483b/op-node/rollup/derive/types.go#L54), so better use `Brotli9` here so it's also covered, and we don't have duplicate coverage of 10.